sklearnex/preview/linear_model/ridge.py

```diff
@@ -0,0 +1,375 @@
+            }
+            if sklearn_check_version("1.2"):
+                X, y = self._validate_data(**check_params)
+                self._validate_params()
+            else:
+                X, y = check_X_y(**check_params)
+
+            if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+                self._normalize = _deprecate_normalize(
+                    self.normalize,
+                    default=False,
+                    estimator_name=self.__class__.__name__,
+                )
+
+            self._initialize_onedal_estimator()
+
+            # Falling back to daal4py if the device is CPU since
+            # onedal does not support non-scalars for alpha, thus
+            # should only be used for GPU/CPU with scalar alpha to not limit the functionality
+            cpu_device = queue is None or queue.sycl_device.is_cpu
+            if cpu_device and not _is_numeric_scalar(self.alpha):
+                self._daal_fit(X, y)
+            else:
+                self._onedal_estimator.fit(X, y, queue=queue)
+
+            self._save_attributes()
+
+        def _onedal_predict(self, X, queue=None):
+            if sklearn_check_version("1.0"):
+                X = self._validate_data(X, accept_sparse=False, reset=False)
+
+            if not hasattr(self, "_onedal_estimator"):
+                self._initialize_onedal_estimator()
+                self._onedal_estimator.coef_ = self.coef_
+                self._onedal_estimator.intercept_ = self.intercept_
+
+            res = self._onedal_estimator.predict(X, queue=queue)
+            return res
+
+        def _onedal_score(self, X, y, sample_weight=None, queue=None):
+            return r2_score(
+                y, self._onedal_predict(X, queue=queue), sample_weight=sample_weight
+            )
+
+        @property
+        def coef_(self):
+            return self._coef
+
+        @coef_.setter
+        def coef_(self, value):
+            if hasattr(self, "_onedal_estimator"):
+                self._onedal_estimator.coef_ = value
+                # checking if the model is already fitted and if so, deleting the model
+                if hasattr(self._onedal_estimator, "_onedal_model"):
+                    del self._onedal_estimator._onedal_model
+            self._coef = value
+
+        @property
+        def intercept_(self):
+            return self._intercept
+
+        @intercept_.setter
+        def intercept_(self, value):
+            if hasattr(self, "_onedal_estimator"):
+                self._onedal_estimator.intercept_ = value
+                # checking if the model is already fitted and if so, deleting the model
+                if hasattr(self._onedal_estimator, "_onedal_model"):
+                    del self._onedal_estimator._onedal_model
+            self._intercept = value
+
+        def _save_attributes(self):
+            self.n_features_in_ = self._onedal_estimator.n_features_in_
+            self._sparse = False
```

Comment:
Where this is used?

Reply:
Probably misunderstanding. The question where `_sparse` attr is used?

Reply:
good point, nowhere in particular, inherited it from sklearnex linear regression implementation with an assumption of possible usage elsewhere. removing it if no objections